src/components/refactor/dialogs/shunt-compensator-creation/shunt-compensator-creation-dialog.js

```diff
@@ -50,20 +55,43 @@ const schema = yup
     .shape({
         [EQUIPMENT_ID]: yup.string().required(),
         [EQUIPMENT_NAME]: yup.string(),
-        [MAXIMUM_NUMBER_OF_SECTIONS]: yup
+        [CHARACTERISTICS_CHOICE]: yup.string().required(),
+        [SUSCEPTANCE_PER_SECTION]: yup
             .number()
-            .min(1, 'ShuntCompensatorErrorMaximumLessThanOne')
-            .required(),
-        [CURRENT_NUMBER_OF_SECTIONS]: yup
+            .nullable()
+            .when([CHARACTERISTICS_CHOICE], {
+                is: (reactivePowerControl) =>
+                    reactivePowerControl ===
+                    CHARACTERISTICS_CHOICES.SUSCEPTANCE.id,
+                then: (schema) => schema.required(),
+            }),
+        [SHUNT_COMPENSATOR_TYPE]: yup.string().when([CHARACTERISTICS_CHOICE], {
+            is: (reactivePowerControl) =>
+                reactivePowerControl ===
+                CHARACTERISTICS_CHOICES.Q_AT_NOMINAL_V.id,
+            then: (schema) =>
+                schema
+                    .oneOf([
+                        SHUNT_COMPENSATOR_TYPES.CAPACITOR.id,
+                        SHUNT_COMPENSATOR_TYPES.REACTOR.id,
+                    ])
+                    .required(),
+        }),
+        [Q_AT_NOMINAL_V]: yup
             .number()
-            .min(0, 'ShuntCompensatorErrorCurrentLessThanZero')
-            .max(
-                yup.ref(MAXIMUM_NUMBER_OF_SECTIONS),
-                'ShuntCompensatorErrorCurrentLessThanMaximum'
-            )
-            .required(),
-        [IDENTICAL_SECTIONS]: yup.bool().required(),
-        [SUSCEPTANCE_PER_SECTION]: yup.number().nullable().required(),
+            .nullable()
+            .when([CHARACTERISTICS_CHOICE], {
+                is: (reactivePowerControl) =>
+                    reactivePowerControl ===
+                    CHARACTERISTICS_CHOICES.Q_AT_NOMINAL_V.id,
```

Comment:
I think here we can directly check the value of reactivePowerControl:
.when([CHARACTERISTICS_CHOICE], {
                is: CHARACTERISTICS_CHOICES.Q_AT_NOMINAL_V.id